engineio/examples/client/main.go

```diff
@@ -39,12 +38,12 @@ func main() {
 		fmt.Println(string(data[:n]))
 	})
 
-	client.OnError(func(ctx *engine.Context, err error) {
+	client.OnError(func(ctx *engineio.Context, err error) {
 		log.Printf("engineio sid %s got error: %s", ctx.Session.ID(), err)
 		ctx.Session.Close()
```

Comment:
is it necessary?

Reply:
Yes. It's explicit.

Reply:
what do to if i wanna keep alive connection after error happened?

Reply:
It depends on the framework. Some errors could resume, but some are not. We could add a resumable error if necessary and try to cast in `OnError`. If an error could cast to this resumable error, then it could keep alive. See [Temporary()](https://pkg.go.dev/net#Error).

Reply:
> If an error could cast to this resumable error, then it could keep alive. See

it is important thing. i think is it worthy of comment.

Reply:
Added `Error` type.

Reply:
What do you think about support similar signatures by `On` methods?

Reply:
probably, hide `error` in `OnError` and  `io.Reader` in `OnMessage` handlers?

Reply:
`OnMessage()` always receives packets with type message, so `io.Reader` is enough.

`OnError()` is another thing. The error is not from packets, but generating while processing packets. I don't understand what to hide here.

Reply:
@mrfoe7 Do you still have questions for this thread?

Reply:
Nope, looks good